include/tsl/ordered_hash.h

```diff
@@ -1098,6 +1098,44 @@ class ordered_hash : private Hash, private KeyEqual {
     return 1;
   }
 
+  /**
+   * Clear a bucket without touching the container holding the values.
+   */
+  void clear_bucket(typename buckets_container_type::iterator it) {
+    tsl_oh_assert(it != m_buckets_data.end());
+    it->clear();
+    backward_shift(std::size_t(std::distance(m_buckets_data.begin(), it)));
+  }
+
+  /**
+   * Remove all entries for which the given predicate matches.
+   */
+  template <class Predicate>
+  size_type erase_if(Predicate &pred) {
+    using const_ref = typename values_container_type::const_reference;
+    auto last = m_values.end();
```

Comment:
I `last` doesn't change, I would add a `const` for clarity. And eventually just use `m_values.end()`.

Reply:
Since `m_values.end()` is used in three places, I made it const.